spec/models/with_gamification.rb

```diff
@@ -0,0 +1,90 @@
+require 'spec_helper'
+
+describe WithGamification, organization_workspace: :test do
+
+  def award_experience_points!(assignment, status)
+    assignment.status = status
+    assignment.award_experience_points!
+    assignment.update_top_submission!
+  end
```

Comment:
This is not ideal because it depends on order: points must be awarded before updating the top submission, hence this helper method.

One way of avoiding this would be updating the top submission _inside_ `award_experience_points!` but it doesn't feel right to me.

Right now it works like this https://github.com/mumuki/mumuki-domain/blob/78209fb65cc4ffebb137987956ab83b20804b47f/lib/mumuki/domain/submission/base.rb#L48-L54

but the order of execution is not tested, as points awarding and submission updating are tested individually.

One possible option to enforce this order would be also testing experience points on this set of specs

https://github.com/mumuki/mumuki-domain/blob/c31a9c3a65c9a3017f026ec2fe086c97da6bb55d/spec/models/assignment_spec.rb#L369

which already mimic submitting a solution (therefore going through `save_results!`), but I thought it'd get a little messy - too many things at once. I'm hearing opinions.

Reply:
I would prefer to use the second option. 
Adding that helper method may cause some scenarios to pass where in reality they don't.

You could test this in the other spec and remove this one if you think it's less messy :smile:

Reply:
I'm not 100% sure where I like it the most. I'll merge them in one single spec file, I'll think which one.